### ISSUE TITLE
How can i get publish message failed info or a failed callBack？

### ISSUE THREAD

**Issue by GMCoder** [mobile-web-messaging/MQTTKit#46]

**Comment by GMCoder** [comment]
When i publish a message，i want get a failed call back.Just have success call back now.

**Comment by dbburgess** [comment]
I don't think this library supports that. You could set the QoS to something higher to confirm the message was received, but that wouldn't help too much with catching errors sending, unfortunately.

**Comment by GMCoder** [comment]
Thank you，I think solve this problem should be change other library.Can you give me some advice？

**Comment by dbburgess** [comment]
You're welcome. I'm not sure I understand. Do you want to try to modify this library to support it?

**Comment by GMCoder** [comment]
Yes,how can i modify this library to support it?

**Comment by dbburgess** [comment]
This project is [no longer maintained](http://jmesnil.net/weblog/2015/09/04/stepping-out-from-personal-open-source-projects/), but you are free to fork it and modify it to suit your needs.

The relevant section is [here](https://github.com/mobile-web-messaging/MQTTKit/blob/master/MQTTKit/MQTTKit.m#L311), where the `publishData` method is. It's just a wrapper around [mosquitto](http://mosquitto.org/), so you can look there for more details. The [publish method](http://mosquitto.org/api/files/mosquitto-h.html#mosquitto_publish) being used _does_ actually return values that could be used for at least _some_ error handling.

Those links should help you get started. Beyond that, I probably can't be of much help. Good luck!

**Comment by GMCoder** [comment]
Thank you!I am so grateful to you.
